python/cherry_core/ingest/__init__.py

```diff
@@ -25,17 +25,17 @@ class Query:
 
 @dataclass
 class ProviderConfig:
-    kind: ProviderKind
-    url: Optional[str] = None
-    bearer_token: Optional[str] = None
-    max_num_retries: Optional[int] = None
-    retry_backoff_ms: Optional[int] = None
-    retry_base_ms: Optional[int] = None
-    retry_ceiling_ms: Optional[int] = None
-    req_timeout_millis: Optional[int] = None
-    stop_on_head: bool = False
-    head_poll_interval_millis: Optional[int] = None
-    buffer_size: Optional[int] = None
+    kind: ProviderKind # (Required) The selected provider.
+    url: Optional[str] = None # (Optional) The URL of the provider.
+    bearer_token: Optional[str] = None # (Optional) Optional authentication token for protected APIs.
+    max_num_retries: Optional[int] = None # (Optional) Maximum number of retries for failed requests.
+    retry_backoff_ms: Optional[int] = None # (Optional) Delay between retries in milliseconds, increases with each attempt.
+    retry_base_ms: Optional[int] = None # (Optional) Base retry delay in milliseconds.
+    retry_ceiling_ms: Optional[int] = None # (Optional) Maximum retry delay in milliseconds.
+    req_timeout_millis: Optional[int] = None # (Optional) Request timeout in milliseconds.
+    stop_on_head: bool = False # (Optional) Whether to automatically stop when reaching the blockchain head or keep the pipeline running indefinitely.
+    head_poll_interval_millis: Optional[int] = None # (Optional) How frequently (in milliseconds) to poll the blockchain head for updates.
+    buffer_size: Optional[int] = None # (Optional) Internal buffer size for streaming data efficiently.
```

Comment:
this determines how many responses are queued before the ingestion stops and waits for caller to consume requests from the queue